Refinery: add new method for 'less than or equal' integer inputs. Add…

lukastocker opened the pull request:
… a new test and language variable for error message.

lukastocker committed:
Refinery: add new method for 'less than or equal' integer inputs. Add a new test and language variable for error message.

klees commented on src/Refinery/Integer/Group.php:
```diff
@@ -55,4 +55,16 @@ public function isGreaterThanOrEqual(int $minimum) : GreaterThanOrEqual
+     * less than or equal the defined upper limit.
+     *
+     * @param int $maximum
+     * @return LessThanOrEqual
```
Docstrings can be removed if the typehint give as much information.

klees commented on src/Refinery/Integer/LessThanOrEqual.php:
```diff
@@ -0,0 +1,32 @@
+    /**
+     * @var int
+     */
+    protected $max;
```
```suggestion
    protected int $max;
```
... and remove docstrings. Yay, PHP 7.4.

klees reviewed (approved):
LGTM in general, 2 minor change requests.

@mjansenDatabay Would you please review as well and, in case, press the button? Thx!

mjansenDatabay reviewed (changes requested):
Thx @lukastocker again for a new `Refinery` transformation.

I don't have anything to add. Please change the findings @klees mentioned. IMO this is ready to merge afterwards.

lukastocker commented:
Hi @mjansenDatabay and @klees ,

thanks for your quick reviews. I just implemented and pushed the changes.
As for the redundant docstrings in the Group.php file: I also removed the docstring for "isGreaterThan", "isLessThan" and "isGreaterThanOrEqual" to standardize the coding style.

Kind regards
Luka

mjansenDatabay commented:
Thanks a lot @lukastocker , I just started a "Streamlining Removal of PHPDocs Feature Branch" ;).

mjansenDatabay commented on src/Refinery/Integer/LessThanOrEqual.php:
```diff
@@ -0,0 +1,29 @@
+    {
+        $this->max = $max;
+        parent::__construct(
+            function ($value) {
```
Maybe we could add a return type declaration here. I guess this could also be streamlined for other constraints.

mjansenDatabay commented on src/Refinery/Integer/LessThanOrEqual.php:
```diff
@@ -0,0 +1,29 @@
+            function ($value) {
+                return $value <= $this->max;
+            },
+            function ($txt, $value) {
```
Maybe we could add a return type declaration here (and a type hint for $txt). I guess this could also be streamlined for other constraints.

mjansenDatabay reviewed (changes requested):
@lukastocker Could you please add the return types/type hints for the functions (where appropriate) in THIS contraint impl.? I'll merge this PR aftwards.

lukastocker commented:
Hi @mjansenDatabay,

the typehints are implemented. 
For the other constraints I will open a new PR with a appropriate title/description.

Kind regards,
Luka Stocker

mjansenDatabay commented:
> For the other constraints I will open a new PR with a appropriate title/description.

Wow, thanks a lot @lukastocker !
